### ISSUE TITLE
In master on Windows, memory usage increases linearly over time for both MS Basic and Muse Sounds, at different rates

### ISSUE THREAD

**Issue by metasekk** [musescore/MuseScore#22721]
### Issue type

General playback bug

### Bug description

Closing and reopening a score leads to a gradual (affine) increase in the RAM memory used by MuseScore, with very rapid growth for Muse Sounds, as though the sound banks loaded in memory were never released, but instead, stacked.

### Steps to reproduce

1. For Windows users, open the task manager.
2. Open [this score](https://github.com/musescore/MuseScore/files/15248035/Memory_usage_MuseSounds.zip) for Muse Sounds (**empty score** with Muse Sounds as the default profile) or [this one](https://github.com/musescore/MuseScore/files/15248034/Memory_usage_MSBasic.zip) for MS Basic (same score with MS Basic as the default profile).
3. Close it and reopen it several times, and compare the memory of the task "MuseScore4.exe" after each cycle.

I provide a (tedious) video [here](https://drive.google.com/file/d/1FE5QlYmy6XjAOoYqqYwDa2Qj0Hk0ODge/view?usp=sharing), demonstrating this process step by step (beware: I couldn't change the language of the task manager, therefore it is in French; memory is the column entitled "Mémoire" - a quite transparent one). The results are rendered quantitative below.

### Screenshots/Screen recordings

I conducted two studies of repeated closing/reopening cycles (three times each), for MS Basic and MuseSounds respectively. Here are my findings:

For Muse Sounds :

![Average_MS4_Memory_for_MuseSounds_3_studies](https://github.com/musescore/MuseScore/assets/120221700/0341bef6-f2f5-496c-8598-baf63a0f0224)

* The figures are highly reproducible : the standard deviation for each figure is <5 Mo.
* It increases monotonically : no drop nor jump occured in the three cases.

For MS Basic :

![Average_MS4_Memory_for_MSBasic_3_studies](https://github.com/musescore/MuseScore/assets/120221700/c55235dd-4f3e-4c23-97d0-ff6a6f23f4be)

* The figures are also highly reproducible on my side : the maximum standard deviation for each figure is <20 Mo.
* A very surprising "jump" from ~350 Mo to ~800 Mo can be seen for MS Basic. It was observed in the three studies, and every time between 5 and 10 re-opening cycles. In 1 case (over 3), there was another drop between 15-20 reopenings, where the memory went back to ~400 Mo. 

I don't know what it reveals about the way the memory allocation/deallocation is implemented, but based on this, I suspect it is different between MS Basic and Muse Sounds, which was quite surprising to me. For MS Basic, the jumps and drops in memory might indicate something dubious in the handling of it.

### MuseScore Version
The results above were obtained in 4.4 master (tested on MuseScoreNightly-241290306-master-48c7fdd-x86_64), but the issue **occurs identically in 4.3**

**Already occurs in 4.2.1 and 4.1.0**, but at a much slower growth rate than in 4.4 and 4.3. 
Occurs similarly with **all versions of MuseSampler** (tested with **MuseSampler 0.6.3, 0.6.1, 0.5.1, 0.3.2**)

### Regression

I don't know

### Operating system

Win10 (22H2); RAM: 16Go; CPU: Intel Core i5-9300H (2.40GHz)

### Additional context

NB1: I could only conduct this experiment on Windows 10.
NB2: On "real" large scores, the amount of RAM required by one score (when first opened) can climb up to 7 Go, making this issue immediately critical when trying to reopen the score only once. 

Somewhat related to #22094, #20889 and possibly #22710; part of #23112.

**Comment by bebobebo** [comment]
I'm working through a set of 10 related piano scores. Because I want the information in the title frame to be identically formatted across them, my process is to
1. Complete score, play through once to pick up any audible errors, and export to PDF
2. "Save as" to next score, edit the title frame. Delete all but the first measure and re-add the desired number of measures in., 
3. Repeat from 1.

The scores are not that big, typically 100 measures or less solo piano. No insertions of images.

After a few hours of this, my Windows 11 memory usage for Musecore was about 3.7GB for that one open score. Everything was going to a crawl. I was not even able to close Musescore and had to use End Task to terminate the process. After re-opening the score, the memory usage wwas about 415MB

OS: Windows 11 Version 2009 or later, Arch.: x86_64, MuseScore Studio version (64-bit): 4.4.4-243461245, revision: github-musescore-musescore-2232670

Device i7, 16GB

**Comment by FrancRos31** [comment]
@bebobebo it's better to open a new issue, or add your comment in a report already open.

**Comment by bebobebo** [comment]
Not every Musescore user is a github expert. Musescore needs to decide how to accept typical user reports given that it doesn't like bug reports on the forums.  If it just likes to make it difficult to submit bugs, then it won't get as many as it needs. It's a poor philosophy.

**Comment by FrancRos31** [comment]
Well, no one is an expert of any new site, you have to practice its functioning. Many times a report on Github is opened after a bug report on the forum. 
I suggested you to open a new issue (or to search a report still open) because your request would be more discoverable, being a fresh one, that's it.

**Comment by MarcSabatella** [comment]
> Not every Musescore user is a github expert. Musescore needs to decide how to accept typical user reports given that it doesn't like bug reports on the forums. If it just likes to make it difficult to submit bugs, then it won't get as many as it needs. It's a poor philosophy.

You don't need to be an "expert".  Simply click "New issue" button at the top of the page, then follow the prompts from there.  It's easily within the grasp of anyone capable of using MuseScore in the first place!

FWIW, though, I'm not sure what you are experiencing *is* a new issue - could well be a symptom of the same thing described here.  So until it's clear that there truly is something different going on, I'm not sure it's wroth creating a new issue.
